spec/fixtures/middleware/internal_info_setter.rb

```diff
@@ -0,0 +1,11 @@
+class InternalInfoSetter
+  MESSAGE = "set by internal_middleware"
+  def initialize(bugsnag)
+    @bugsnag = bugsnag
+  end
+
+  def call(notification)
+    notification.info = MESSAGE
```

Comment:
This was my lack of understanding - I was attempting to pass do...end style blocks to have_sent_notification (rather than curly brace style blocks) so these tests were passing when they shouldn't have been.